tests/embind/embind.test.js

```diff
@@ -378,20 +378,38 @@ module({
     });
 
     BaseFixture.extend("string", function() {
+        var embindStdStringUTF8Support = (Module['EMBIND_STD_STRING_UTF8_SUPPORT'] == true);
+
         test("non-ascii strings", function() {
-            var expected = '';
-            for (var i = 0; i < 128; ++i) {
-                expected += String.fromCharCode(128 + i);
+
+            if(embindStdStringUTF8Support) {
+                //ASCII
+                var expected = 'aei';
+                //Latin-1 Supplement
+                expected += '\u00E1\u00E9\u00ED';
+                //Greek
+                expected += '\u03B1\u03B5\u03B9';
+                //Cyrillic
+                expected += '\u0416\u041B\u0424';
+                //CJK
+                expected += '\u5F9E\u7345\u5B50';
+                //Euro sign
+                expected += '\u20AC';
```

Comment:
I'd like to see a test that codifies the behavior when invalid UTF-8 data is given from the C++ side.